src/core/lib/channel/context.h

```diff
@@ -32,6 +32,9 @@ typedef enum {
   /// Value is a \a census_context.
   GRPC_CONTEXT_TRACING,
 
+  /// Value is a CallTracer object.
+  GRPC_CONTEXT_CALL_TRACER,
```

Comment:
We might be able to just reuse `GRPC_CONTEXT_TRACING` for this.

Reply:
Or alternatively, if we don't want to overload the same `void *` for different types in different parts of the code, we could keep both for now, and delete GRPC_CONTEXT_TRACING later.

Reply:
I don't think that makes sense, for two reasons.

First, `GRPC_CONTEXT_TRACING` is used for passing the context between the surface and the census filter, and we still need to support that.  While the census filter probably should grab what's in that context just once when it is initialized and not need it afterward, there really isn't any guarantee that it will do so; the census context is actually supposed to remain available to it at any time during the existence of the call.  Also, I believe that there is a C-core surface API that exposes the census context to the wrapped language level, and it probably needs to be able to call it at any time during the call as well.

And second, reusing `GRPC_CONTEXT_TRACING` does not seem safe, because any code looking at that entry will expect to be able to cast its value to a `CensusContext` object.  If we replace that value with a different type of object, that will lead to crashes.

Reply:
What we expose to the wrapped layer is just an opaque `struct census_context` that we cast to around to a real type. We can just replace that with a CallTracer object.

Reply:
I'd rather not change the API used by wrapped languages until we have an approved design for the longer-term stats changes.  For now, this API is needed only between the census filter and code below it, so let's not expose it publicly if we don't have to.